hammerblade/torch/kernel/kernel_conv.cpp

```diff
@@ -129,7 +129,7 @@ extern "C" {
     bsg_cuda_print_stat_kernel_start();
 
     // init input grads
-    hb_tiled_foreach([]() {return 0.0;}, x);
+    hb_tiled_foreach(x, []() {return 0.0;});
```

Comment:
Ok for this syntax we need @vb000 's input. So @vb000 , do you have to put the lambda at the very beginning for your code to work?

Reply:
No, I was trying to make this work with variadic arguments to avoid repeating the implementation, in which case, tensor arguments have to be in the end. For example,

```
hb_tiled_foreach(<lambda>, T... tensor_args)
```

is supposed to handle all variants. `tensor_args` is a parameter pack here, and to achieve this with `hb_tiled_foreach`, we need to iterate over each argument in the pack and apply a computation. But in C++11, there doesn't seem to be a way to do compile time loops on parameter packs. So, I couldn't get this done. But I moved tensor arguments to the end, in case any meta-programming wizard wants to tackle this ;). We can move lambda to the end if that's convenient for any of your implementations.

Reply:
Yes, I tried using variadic arguments for that too, but it seemed like there would still have to be different implementations for different operations on some level so I worked with the overloaded function itself. However, my implementation is indifferent to the position of the lambda. I think @cbatten preferred having the lambda at the end.